sourced/ml/algorithms/token_parser.py

```diff
@@ -1,29 +1,50 @@
+from enum import Enum
 import re
 
 import Stemmer
 
 
+class TokenMeta(Enum):
+    """Metadata that should allow to reconstruct initial identifier from a list of tokens."""
+    DELIMITER = 1
+    TOKEN_UPPER = 2
+    TOKEN_LOWER = 3
+    TOKEN_STARTS_WITH_UPPER = 4
```

Comment:
```suggestion
    TOKEN_CAPITALIZE = 4
```
It corresponds to `"abc".capitalize()`